distsql/request_builder.go

```diff
@@ -165,7 +166,11 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 	builder.Request.NotFillCache = sv.StmtCtx.NotFillCache
 	builder.Request.Priority = builder.getKVPriority(sv)
 	builder.Request.ReplicaRead = sv.GetReplicaRead()
-	builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
+	if sv.SnapshotInfoschema != nil {
+		builder.Request.SchemaVar = sv.SnapshotInfoschema.(infoschema.InfoSchema).SchemaMetaVersion()
```

Comment:
Could you add an integration test for it?

Reply:
SchemaVar is only checked by TiFlash. None way to integration test.

Reply:
Is it better use [infoschema.GetInfoSchema](https://github.com/pingcap/tidb/blob/9f0736e/infoschema/infoschema.go#L341) here?